diff -urN a/gvsbuild/projects.py b/gvsbuild/projects.py
--- a/gvsbuild/projects.py	2020-04-20 18:10:24.526745700 +0200
+++ b/gvsbuild/projects.py	2020-04-29 14:55:19.453130500 +0200
@@ -820,7 +820,7 @@
         self.exec_cmd(r'%(gtk_dir)s\bin\gtk-update-icon-cache.exe --ignore-theme-index --force "%(gtk_dir)s\share\icons\hicolor"')
 
 @project_add
-class Project_gtk3_24(Tarball, Meson):
+class Project_gtk3_24(GitRepo, Meson):
     def __init__(self):
         if self.opts.gtk3_ver != '3.24':
             self.ignore()
@@ -829,11 +829,13 @@
         Project.__init__(self,
             'gtk3',
             prj_dir='gtk3-24',
-            archive_url = 'http://ftp.acc.umu.se/pub/GNOME/sources/gtk+/3.24/gtk+-3.24.17.tar.xz',
-            hash = 'f210255b221cb0f0db3e7b21399983b715c9dda6eb1e5c2f7fdf38f4f1b6bac0',
+            repo_url = 'https://gitlab.gnome.org/GNOME/gtk.git',
+            fetch_submodules = False,
+            tag = 'gtk-3-24',
             dependencies = ['atk', 'gdk-pixbuf', 'pango', 'libepoxy'],
             patches = [
                 'gtk_update_icon_cache.patch',
+                'win32.patch',
             ],
             )
         if self.opts.enable_gi:
diff -urN a/patches/gtk3-24/win32.patch b/patches/gtk3-24/win32.patch
--- a/patches/gtk3-24/win32.patch	1970-01-01 01:00:00.000000000 +0100
+++ b/patches/gtk3-24/win32.patch	2020-04-28 15:30:32.448803900 +0200
@@ -0,0 +1,12 @@
+diff -ur a/gtk/theme/win32/gtk-win32-base.css b/gtk/theme/win32/gtk-win32-base.css
+--- a/gtk/theme/win32/gtk-win32-base.css	2020-04-17 18:31:33.000000000 +0200
++++ b/gtk/theme/win32/gtk-win32-base.css	2020-04-18 15:23:47.527868200 +0200
+@@ -98,7 +98,7 @@
+ 
+ .view:selected {
+   background-color: shade(-gtk-win32-color(edit, window), 0.9);
+-  color: @fg_color;
++  color: #000;
+ }
+ 
+ .view:selected:focus {
