--- a/client.py	2019-12-08 01:22:54.177670900 +0100
+++ b/client.py	2019-11-26 16:28:51.878474100 +0100
@@ -668,10 +668,8 @@
             bool: True is successfully started the daemon, False otherwise.
 
         """
-        # subprocess.popen does not work with unicode args (with non-ascii characters) on windows
-        config = config.encode(sys.getfilesystemencoding())
         try:
-            subprocess.Popen(['deluged', '--port=%s' % port, b'--config=%s' % config])
+            subprocess.Popen(['deluged', '--port=%s' % port, '--config=%s' % config])
         except OSError as ex:
             from errno import ENOENT
 
